sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql

```diff
@@ -0,0 +1,643 @@
+    IF(
+      ANY_VALUE(submission_date HAVING MIN submission_date) >= '2018-10-30',
+      MIN(submission_timestamp_min),
+      TIMESTAMP(ANY_VALUE(submission_date HAVING MIN submission_date))
+    ) AS first_seen_date,
+    TIMESTAMP(
+      ARRAY_AGG(submission_date IGNORE NULLS ORDER BY submission_date)[SAFE_OFFSET(1)]
+    ) AS second_seen_date,
+    CAST(NULL AS STRING) AS architecture,
+    ANY_VALUE(app_build_id HAVING MIN submission_date) AS app_build_id,
+    ANY_VALUE(app_name HAVING MIN submission_date) AS app_name,
+    CAST(NULL AS STRING) AS platform_version,
+    ANY_VALUE(vendor HAVING MIN submission_date) AS vendor,
+    ANY_VALUE(app_version HAVING MIN submission_date) AS app_version,
+    CAST(NULL AS STRING) AS xpcom_abi,
+    CAST(NULL AS STRING) AS document_id,
+    ANY_VALUE(distribution_id HAVING MIN submission_date) AS distribution_id,
+    CAST(NULL AS STRING) AS partner_distribution_version,
+    CAST(NULL AS STRING) AS partner_distributor,
+    CAST(NULL AS STRING) AS partner_distributor_channel,
+    CAST(NULL AS STRING) AS partner_id,
+    ANY_VALUE(attribution.campaign HAVING MIN submission_date) AS attribution_campaign,
+    ANY_VALUE(attribution.content HAVING MIN submission_date) AS attribution_content,
+    ANY_VALUE(attribution.experiment HAVING MIN submission_date) AS attribution_experiment,
+    ANY_VALUE(attribution.medium HAVING MIN submission_date) AS attribution_medium,
+    ANY_VALUE(attribution.source HAVING MIN submission_date) AS attribution_source,
+    ANY_VALUE(
+      default_search_engine_data_load_path
+      HAVING
+        MIN submission_date
+    ) AS engine_data_load_path,
+    ANY_VALUE(default_search_engine_data_name HAVING MIN submission_date) AS engine_data_name,
+    ANY_VALUE(default_search_engine_data_origin HAVING MIN submission_date) AS engine_data_origin,
+    ANY_VALUE(
+      default_search_engine_data_submission_url
+      HAVING
+        MIN submission_date
+    ) AS engine_data_submission_url,
+    CAST(NULL AS STRING) AS apple_model_id,
+    ANY_VALUE(city HAVING MIN submission_date) AS city,
+    CAST(NULL AS STRING) AS db_version,
+    ANY_VALUE(geo_subdivision1 HAVING MIN submission_date) AS subdivision1,
+    ANY_VALUE(normalized_channel HAVING MIN submission_date) AS normalized_channel,
+    ANY_VALUE(country HAVING MIN submission_date) AS country,
+    ANY_VALUE(os HAVING MIN submission_date) AS normalized_os,
+    ANY_VALUE(normalized_os_version HAVING MIN submission_date) AS normalized_os_version,
+    CAST(NULL AS STRING) AS startup_profile_selection_reason,
+    ANY_VALUE(attribution.dltoken HAVING MIN submission_date) AS attribution_dltoken,
+    CAST(NULL AS STRING) AS attribution_dlsource
+  FROM
+    `moz-fx-data-shared-prod.telemetry_derived.clients_daily_v6`
+  WHERE
+    {% if is_init() %}
+      submission_date >= '2010-01-01'
+      {% if parallel_run() %}
+        AND sample_id = {sample_id}
+      {% endif %}
+    {% else %}
```

Comment:
Running sample_ids (0,1) in parallel, each sample_id [runs in avg. 19 min and processes 900 GB](https://console.cloud.google.com/bigquery?sq=470293616514:4d738027e7a546488a24c89650f41f8b&project=mozdata) so I think it would be okay, I guess we need to run the 100 sample_ids for this validation.

Reply:
@fbertsch , is that the size for the main ping table (`telemetry.main`), or client daily (`telemetry.clients_daily`)?Currently, this code is pulling the data from clients daily, not main pings.